exp/sampler.jl

```diff
@@ -61,6 +85,7 @@ function build_and_solve_model(data, config, dataset_name)
     )
 
     set_silent(opf.model)
+    set_time_limit(opf.model, config, dataset_name)
```

Comment:
This will only set a time limit if the user didn't already set one (see [JuMP.time_limit_sec](https://jump.dev/JuMP.jl/stable/api/JuMP/#JuMP.time_limit_sec)).
```suggestion
    # Set time limit if one is not already set
    if JuMP.time_limit_sec(opf.model) !== nothing
        JuMP.set_time_limit_sec(opf.model, time_limit)
    end
```


Note that `JuMP.set_time_limit_sec(model, nothing)` removes any solver time limit, hence:
* if the user set a solver-level time limit, we don't over-ride it
* if the user didn't set a solver time limit, we set it to `time_limit`, in which case either
  * `time_limit` is `nothing`, and we haven't changed anything, or
  * `time_limit` is not `nothing`, and we set a time limit to a non-zero value

Reply:
should this be `===` instead of `!==`?

Reply:
also `time_limit_sec` seems to sometimes return `Inf`...